wgpu-core/src/adapter.rs

```diff
@@ -0,0 +1,246 @@
+use crate::{
+    backend,
+    hub::{GfxBackend, Global, IdentityFilter, Token},
+    id::AdapterId,
+    Backend,
+};
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+pub use hal::adapter::AdapterInfo;
+use hal::Instance;
+
+#[derive(Debug)]
+pub struct RawAdapter<B: hal::Backend> {
+    pub(crate) raw: hal::adapter::Adapter<B>,
+}
+
+#[allow(clippy::large_enum_variant)]
```

Comment:
Clippy only complains because of `gfx_backend_dx11::Backend`, I could easily put this in a Box, but I'm not entirely sure what this struct entails.